ucp/benchmarks/send_recv.py

```diff
@@ -77,43 +63,16 @@ def server(queue, args):
         xp.cuda.runtime.setDevice(args.server_dev)
         xp.cuda.set_allocator(rmm.rmm_cupy_allocator)
 
-    ucp.init()
-
-    register_am_allocators(args)
-
-    async def run():
-        async def server_handler(ep):
-
-            if not args.enable_am:
-                msg_recv_list = []
-                if not args.reuse_alloc:
-                    for _ in range(args.n_iter + args.n_warmup_iter):
-                        msg_recv_list.append(xp.zeros(args.n_bytes, dtype="u1"))
-                else:
-                    t = xp.zeros(args.n_bytes, dtype="u1")
-                    for _ in range(args.n_iter + args.n_warmup_iter):
-                        msg_recv_list.append(t)
-
-                assert msg_recv_list[0].nbytes == args.n_bytes
+    if args.backend == "ucp-async":
+        server = UCXPyAsyncServer(args, xp, queue)
+    elif args.backend == "ucp-core":
+        server = UCXPyCoreServer(args, xp, queue)
 
-            for i in range(args.n_iter + args.n_warmup_iter):
-                if args.enable_am is True:
-                    recv = await ep.am_recv()
-                    await ep.am_send(recv)
-                else:
-                    await ep.recv(msg_recv_list[i])
-                    await ep.send(msg_recv_list[i])
-            await ep.close()
-            lf.close()
-
-        lf = ucp.create_listener(server_handler, port=args.port)
-        queue.put(lf.port)
-
-        while not lf.closed():
-            await asyncio.sleep(0.5)
-
-    loop = get_event_loop()
-    loop.run_until_complete(run())
+    if asyncio.iscoroutinefunction(server.run):
+        loop = get_event_loop()
+        loop.run_until_complete(server.run())
+    else:
+        server.run()
```

Comment:
Maybe merge the conditions (since `server.run` is a coroutine iff `args.backend == "ucxpy-async"`) ?

```
if args.backend == "ucxpy-async":
    server = UCXPyAsyncServer(...)
    get_event_loop().run_until_complete(server.run())
elif ...:
    server = UCXPyCoreServer(...)
    server.run()
```
?

Reply:
This iff is only temporary, in https://github.com/rapidsai/ucx-py/pull/874 we introduce another backend that makes this a more logical condition: https://github.com/rapidsai/ucx-py/pull/874/files#diff-e2b415b95132fb557a1c441b2959d7da1b114a8164eac1156161d3101eb376a3R85-R91 . So if you don't mind, I'll disregard this suggestion and we can make any changes in that follow-up PR.